test/files/neg/named-booleans.scala

```diff
@@ -0,0 +1,47 @@
+// scalac: -Werror -Xlint:named-booleans
+
+class C {
+  def f(n: Int = 42, x: Boolean, y: Boolean) = if (x && y) n else 0
+
+  def g(x: Any) =
+    x match {
+      case (true, false) => 0
+      case _ => 1
+    }
+  var b = false
+}
+
+object Test extends App {
+  val c = new C
+  val b = false
+  val x0 = c.f(17, true, b) // warn
+  val x1 = c.f(17, x = true, b) // nowarn
+  val x2 = c.f(y = b, n = 17, x = true) // nowarn
+  c.b = true
+  val y = Some(false)
+  val z = Option(false)
+  val w = (true, false)
+  val v = c g true  // nowarn?
```

Comment:
This probably takes a different code path in the parser, see `makeBinop`

Reply:
that's right. I decided it's ok because whatever. I did see the idiom in the codebase!